components/movementsensor/gpsnmea/serial.go

```diff
@@ -95,33 +91,25 @@ func (g *SerialNMEAMovementSensor) Start(ctx context.Context) error {
 	g.activeBackgroundWorkers.Add(1)
 	utils.PanicCapturingGo(func() {
 		defer g.activeBackgroundWorkers.Done()
-		r := bufio.NewReader(g.dev)
 		for {
 			select {
```

Comment:
Having one select statement for everything isn't good enough: if multiple cases are satisfied, there's no guarantee on which one will fire, and perhaps we'll read lines forever without noticing that we should shut down. Have a second, non-blocking select statement for just the context.